src/components/productList/ProductListFlatList.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import {
+  ActivityIndicator,
+  FlatList,
+  RefreshControl,
+  Text,
+  View,
+} from 'react-native';
+import {styles} from './styles';
+import {COLORS} from '../../utils/colors';
+import ProductListCard from '../productListCard';
+import STRING_UTILS from '../../utils/StringUtils';
+import {getIncludedImageById} from '../../api/ImageApi';
+import {useNavigation} from '@react-navigation/native';
+
+type ProductListFlatListProp = {
+  data: {};
+  included: {};
+  onRefresh: any;
+  isRefreshing: boolean;
+};
+const ProductListFlatList = ({
+  data,
+  included,
+  onRefresh,
+  isRefreshing,
+}: ProductListFlatListProp) => {
+  const navigation = useNavigation();
+  if (!data) {
+    return (
+      <View style={[styles.onLoadDataContainer, styles.onLoadDataHorizontal]}>
+        <ActivityIndicator size="large" color={COLORS.blue_500} />
+        <View>
+          <Text style={styles.loadingData}>Loading...</Text>
+        </View>
+      </View>
+    );
+  } else {
+    return (
+      <FlatList
+        data={data}
+        numColumns={2}
+        onRefresh={onRefresh}
+        refreshControl={
```

Comment:
Passing react element as a child is also a complex behaviour, which must be memoised as well via `useMemo`